File: Algorithms/p014_Longest_Common_Prefix/p014_Longest_Common_Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";
        string s = strs[0];
        for(int i=1;i<strs.size();++i){
            int j = 0;
            while(j<strs[i].size()&&j<s.size()&&strs[i][j]==s[j]) ++j;
            s.resize(j);
            if(s.empty()) return "";
        }
        return s;
    }
};

File: Algorithms/p014_Longest_Common_Prefix/p014_Longest_Common_Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ''
        s = strs[0]
        for i in range(1,len(strs)):
            j = 0
            while j<len(strs[i]) and j<len(s) and strs[i][j]==s[j]:
                j += 1
            s = s[:j]
            if not s:
                return s
        return s
